modules/lang-painless/src/main/java/org/elasticsearch/painless/Compiler.java

```diff
@@ -135,41 +152,32 @@ int newLambdaIdentifier() {
             clazz.getField("$SOURCE").set(null, source);
             clazz.getField("$STATEMENTS").set(null, root.getStatements());
             clazz.getField("$DEFINITION").set(null, definition);
-            java.lang.reflect.Constructor<? extends PainlessScript> constructor = clazz.getConstructor();
 
-            return iface.cast(constructor.newInstance());
+            return clazz.getConstructor();
```

Comment:
Just a note, this will need to change in the future to grab the appropriate constructor given arguments from the script factory?